cf-agent/cf-agent.c

```diff
@@ -238,11 +238,15 @@ int main(int argc, char *argv[])
     {
         Log(LOG_LEVEL_ERR, "CFEngine was not able to get confirmation of promises from cf-promises, so going to failsafe");
         EvalContextClassPutHard(ctx, "failsafe_fallback", "attribute_name=Errors,source=agent");
-        GenericAgentConfigSetInputFile(config, GetInputDir(), "failsafe.cf");
-        policy = LoadPolicy(ctx, config);
+        
+        if (CheckAndGenerateFailsafe(GetInputDir(), "failsafe.cf"))
+        {
+            GenericAgentConfigSetInputFile(config, GetInputDir(), "failsafe.cf");
+            policy = LoadPolicy(ctx, config);
+        }
     }
     assert(policy);
```

Comment:
@pasinskim actually I was wrong that `assert()` is better suited here. `policy` **can** indeed be NULL if `CheckAndGenerateFailsafe()` fails, so your earlier check is better, and `assert()` should be removed.